src/oca_github_bot/tasks/rebase_bot.py

```diff
@@ -0,0 +1,135 @@
+from bs4 import BeautifulSoup
+
+from .. import github
+from ..config import REBASE_BOT_INTRO_MESSAGES, switchable
+from ..manifest import git_modified_addon_dirs, is_maintainer
+from ..queue import getLogger, task
+
+_logger = getLogger(__name__)
+
+
+def _git_call(cmd, cwd="."):
+    subprocess.check_output(
+        cmd, universal_newlines=True, stderr=subprocess.STDOUT, cwd=cwd
+    )
+
+
+def _get_rebase_bot_intro_message():
+    i = random.randint(0, len(REBASE_BOT_INTRO_MESSAGES) - 1)
+    return REBASE_BOT_INTRO_MESSAGES[i]
+
+
+def _user_can_rebase(gh, org, repo, username, addons_dir, target_branch):
+    """
+    Check if a user is allowed to rebase.
+
+    addons_dir must be a git clone of the branch to merge to target_branch.
+    """
+    gh_repo = gh.repository(org, repo)
+    if github.github_user_can_push(gh_repo, username):
+        return True
+    modified_addon_dirs, other_changes = git_modified_addon_dirs(
+        addons_dir, target_branch
+    )
+    if other_changes or not modified_addon_dirs:
+        return False
+    # if we are modifying addons only, then the user must be maintainer of
+    # all of them on the target branch
+    current_branch = github.git_get_current_branch(cwd=addons_dir)
+    try:
+        _git_call(["git", "checkout", target_branch], cwd=addons_dir)
+        return is_maintainer(username, modified_addon_dirs)
+    finally:
+        _git_call(["git", "checkout", current_branch], cwd=addons_dir)
+
+
+@task()
+@switchable("rebase_bot")
+def rebase_bot_start(org, repo, pr, username, dry_run=False, intro_message=None):
+    with github.login() as gh:
+        gh_pr = gh.pull_request(org, repo, pr)
+        target_branch = gh_pr.base.ref
+        try:
+            if not _user_can_rebase(gh, org, repo, username, ".", target_branch):
+                github.gh_call(
+                    gh_pr.create_comment,
+                    f"Sorry @{username} you are not allowed to rebase.\n\n"
+                    f"To do so you must either have push permissions on "
+                    f"the repository, or be a declared maintainer of all "
+                    f"modified addons.\n\n"
+                    f"If you wish to adopt an addon and become it's "
+                    f"[maintainer]"
+                    f"(https://odoo-community.org/page/maintainer-role), "
+                    f"open a pull request to add "
+                    f"your GitHub login to the `maintainers` key of its "
+                    f"manifest.",
+                )
+                return
+            current_branch = github.git_get_current_branch(cwd=".")
+            if not intro_message:
+                intro_message = _get_rebase_bot_intro_message()
+            github.gh_call(
+                gh_pr.create_comment,
+                f"{intro_message}\n"
+                f"Rebasing to [{target_branch}]"
+                f"(https://github.com/{org}/{repo}/pull/{pr}/commits/{current_branch})"
+                f".",
+            )
```

Comment:
Since everything is done in one step there is no need for an intro message. A success or error message is enough.